ros_ws/src/teleoperation/src/joystick_controller.cpp

```diff
@@ -14,8 +17,21 @@ JoystickController::JoystickController()
     this->m_joystick_subscriber =
         this->m_node_handle.subscribe<sensor_msgs::Joy>("joy", 10, &JoystickController::joystickCallback, this);
 
+
+    // get the provided gamepad type
     ros::NodeHandle private_node_handle("~");
-    private_node_handle.getParam(INVERT_STEERING_PARAMETER, this->m_invert_steering);
+    private_node_handle.getParam(GAMEPAD_TYPE_PARAMETER, this->m_gamepad_type);
+
+    // load joystick map for the provided gamepad type
+    ROS_ASSERT_MSG(m_gamepad_type != "xbox360" || m_gamepad_type != "ps3", "Gamepad type should be xbox360 or ps3");
+    if (m_gamepad_type == "xbox360")
+    {
+        m_joystick_map = new JoystickMapXbox360();
+    }
+    else if (m_gamepad_type == "ps3")
+    {
+        m_joystick_map = new JoystickMapPs3();
+    }
```

Comment:
**MAY**
Add an `else` case that throws an exception, outputs an error or gives any kind of indication that something happened that should not have happened.